src/MediaCollections/Filesystem.php

```diff
@@ -133,66 +150,65 @@ public function copyFromMediaLibrary(Media $media, string $targetFile): string
         return $targetFile;
     }
 
-    public function removeAllFiles(Media $media)
+    public function removeAllFiles(Media $media): void
     {
         $mediaDirectory = $this->getMediaDirectory($media);
 
         $conversionsDirectory = $this->getMediaDirectory($media, 'conversions');
 
         $responsiveImagesDirectory = $this->getMediaDirectory($media, 'responsiveImages');
 
-        collect([$mediaDirectory, $conversionsDirectory, $responsiveImagesDirectory])
+        $this->filesystem->disk($media->disk)->deleteDirectory($mediaDirectory);
 
+        collect([$mediaDirectory, $conversionsDirectory, $responsiveImagesDirectory])
             ->each(function ($directory) use ($media) {
-                $this->filesystem->disk($media->disk)->deleteDirectory($directory);
+                $this->filesystem->disk($media->conversions_disk)->deleteDirectory($directory);
             });
```

Comment:
`$mediaDirectory` wordt al verwijderd in die `each`. Lijn 161 mag dus weg denk ik